Noarch package

orbeckst opened the pull request:
Change mdanalysistests into a noarch package (see https://github.com/MDAnalysis/mdanalysis/issues/3727)
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

orbeckst committed:
remove external builds in conda config

orbeckst committed:
incremente build number

orbeckst committed:
only build linux in Azure

orbeckst committed:
removed build requirements from recipe

orbeckst committed:
Merge pull request #1 from MDAnalysis/noarch-package

noarch package

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.


I do have some suggestions for making it better though...


For **recipe**:

 * Whenever possible python packages should use pip. See https://conda-forge.org/docs/maintainer/adding_pkgs.html#use-pip

orbeckst commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 3.21.9, conda-smithy 3.21.0, and conda-forge-pinning 2022.06.27.18.30.08

orbeckst commented:
Comments from PR #32 
- @IAlibay : I believe the way forward in meta.yml is to have noarch in the build section: https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python

orbeckst committed:
declare noarch package

follows https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-python

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found some lint.

Here's what I've got...


For **recipe**:

 * `noarch` packages can't have selectors. If the selectors are necessary, please remove `noarch: python`.
 * noarch: python recipes are required to have a lower bound on the python version. Typically this means putting `python >=3.6` in **both** `host` and `run` but you should check upstream for the package's Python compatibility.

For **recipe**:

 * Whenever possible python packages should use pip. See https://conda-forge.org/docs/maintainer/adding_pkgs.html#use-pip

orbeckst commented on recipe/meta.yaml:
```diff
@@ -12,16 +12,14 @@ source:
-  number: 0
+  number: 1
+  noarch: python
   script: python setup.py install --single-version-externally-managed --record record.txt
```
Can this be rewritten as
```
script: {{ PYTHON }} -m pip install . --no-deps -vv
```

(possibly change requirements.host setuptools --> pip)?

orbeckst replied:
no-deps will not work as we need MDA.

IAlibay replied:
Isn't that already defined under `run`?

orbeckst replied:
Ah, you're right... Do we need the `--single-version-externally-managed --record record.txt` options for something??

orbeckst replied:
I changed it to use pip.

orbeckst committed:
change python restrictions for noarch

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.


I do have some suggestions for making it better though...


For **recipe**:

 * Whenever possible python packages should use pip. See https://conda-forge.org/docs/maintainer/adding_pkgs.html#use-pip

orbeckst commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 3.21.9, conda-smithy 3.21.0, and conda-forge-pinning 2022.06.27.18.30.08

orbeckst commented:
I'll probably close my manual attempt here and let conda-smithy do everything in PR #34 ...

orbeckst commented:
The automated PR #35 really only added noarch but there was other stuff to fix. It looks as if this PR is just as good (and already has more things removed).

@IAlibay perhaps you could have a quick look when you have time?

IAlibay commented:
> The automated PR #35 really only added noarch but there was other stuff to fix. It looks as if this PR is just as good (and already has more things removed).
> 
> @IAlibay perhaps you could have a quick look when you have time?

👍🏽 I'll give it a look through in the morning.

orbeckst committed:
use pip for installation instead of setuptools

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

orbeckst committed:
add pip as host.requirement

Following https://conda-forge.org/docs/maintainer/adding_pkgs.html#use-pip, `pip` should be in `host.requirements`.

IAlibay reviewed (approved):
This seems fine to me, I don't understand thing enough to be 100% sure, but the best thing we can do is test it out and see what happens.
